src/python/pants/pantsd/service/store_gc_service.py

```diff
@@ -22,50 +19,49 @@ class StoreGCService(PantsService):
   performs occasional garbage collection to bound the size of the engine's Store.
   """
 
+  _LEASE_EXTENSION_INTERVAL_SECONDS = 30 * 60
+  _GARBAGE_COLLECTION_INTERVAL_SECONDS = 4 * 60 * 60
+
   def __init__(self, scheduler):
     super(StoreGCService, self).__init__()
     self._scheduler = scheduler
     self._logger = logging.getLogger(__name__)
-    self._lease_extension_thread = None
-    self._garbage_collection_thread = None
 
-  def setup(self, lifecycle_lock, fork_lock):
-    super(StoreGCService, self).setup(lifecycle_lock, fork_lock)
+  @staticmethod
+  def _launch_thread(f):
+    t = threading.Thread(target=f)
+    t.daemon = True
+    t.start()
+    return t
 
   def _extend_lease(self):
-    while True:
-      self._logger.debug("Extending leases")
-      # Grab the fork lock to ensure this thread isn't cloned by a fork while holding the graph
-      # lock.
-      self.fork_lock.acquire()
-      try:
+    while 1:
+      # Use the fork lock to ensure this thread isn't cloned via fork while holding the graph lock.
+      with self.fork_lock:
+        self._logger.debug('Extending leases')
         self._scheduler.lease_files_in_graph()
-      finally:
-        self.fork_lock.release()
-        self._logger.debug("Done extending leases")
-      time.sleep(_LEASE_EXTENSION_INTERVAL_SECONDS)
+        self._logger.debug('Done extending leases')
+      time.sleep(self._LEASE_EXTENSION_INTERVAL_SECONDS)
 
   def _garbage_collect(self):
-    while True:
-      time.sleep(_GARBAGE_COLLECTION_INTERVAL_SECONDS)
+    while 1:
+      time.sleep(self._GARBAGE_COLLECTION_INTERVAL_SECONDS)
```

Comment:
The location of the sleep is inconsistent between the two threads as well... totally doesn't matter... just OCD.

Reply:
yeah, assumed that was intentional as a means to delay the first GC.

Reply:
Yeah, intentional :)